core/domain/activity_jobs_one_off.py

```diff
@@ -471,3 +478,141 @@ def map(snapshot_model):
+        # '<topicId>-<subtopicNum>-<version>'.
+        model_id, version = snapshot_model.id.rsplit('-', 1)
+        model_dict = [
+            {
+                'model': collection_models.CollectionSnapshotMetadataModel,
+                'parent_model': collection_models.CollectionModel,
+                'commit_log_model': (
+                    collection_models.CollectionCommitLogEntryModel),
+                'id_string_format': 'collection-%s-%s'
+            },
+            {
+                'model': exp_models.ExplorationSnapshotMetadataModel,
+                'parent_model': exp_models.ExplorationModel,
+                'commit_log_model': exp_models.ExplorationCommitLogEntryModel,
+                'id_string_format': 'exploration-%s-%s'
+            },
+            {
+                'model': question_models.QuestionSnapshotMetadataModel,
+                'parent_model': question_models.QuestionModel,
+                'commit_log_model': question_models.QuestionCommitLogEntryModel,
+                'id_string_format': 'question-%s-%s'
+            },
+            {
+                'model': skill_models.SkillSnapshotMetadataModel,
+                'parent_model': skill_models.SkillModel,
+                'commit_log_model': skill_models.SkillCommitLogEntryModel,
+                'id_string_format': 'skill-%s-%s'
+            },
+            {
+                'model': story_models.StorySnapshotMetadataModel,
+                'parent_model': story_models.StoryModel,
+                'commit_log_model': story_models.StoryCommitLogEntryModel,
+                'id_string_format': 'story-%s-%s'
+            },
+            {
+                'model': topic_models.TopicSnapshotMetadataModel,
+                'parent_model': topic_models.TopicModel,
+                'commit_log_model': topic_models.TopicCommitLogEntryModel,
+                'id_string_format': 'topic-%s-%s'
+            },
+            {
+                'model': subtopic_models.SubtopicPageSnapshotMetadataModel,
+                'parent_model': subtopic_models.SubtopicPageModel,
+                'commit_log_model': (
+                    subtopic_models.SubtopicPageCommitLogEntryModel),
+                'id_string_format': 'subtopicpage-%s-%s'
+            },
+            {
+                'model': topic_models.TopicRightsSnapshotMetadataModel,
+                'parent_model': topic_models.TopicRightsModel,
+                'commit_log_model': topic_models.TopicCommitLogEntryModel,
+                'id_string_format': 'rights-%s-%s'
+            },
+        ]
+
+        for val in model_dict:
+            if isinstance(snapshot_model, val['model']):
+                commit_log_id = val['id_string_format'] % (model_id, version)
+                parent_model = val['parent_model'].get_by_id(model_id)
+                commit_log_model = val['commit_log_model'].get_by_id(
+                    commit_log_id)
+                break
```

Comment:
Probably better to model the above as a dict keyed by the model class name? Then you don't need to iterate through everything, you can just read off the value.

Reply:
Done.